stats.go

```diff
@@ -95,23 +71,67 @@ func (c *counter) Inc(v int64) {
 func (c *counter) report(name string, tags map[string]string, r StatsReporter) {
 	curr := atomic.LoadInt64(&c.curr)
 
-	prev := c.prev
+	prev := atomic.LoadInt64(&c.prev)
 	if prev == curr {
 		return
 	}
 	atomic.StoreInt64(&c.prev, curr)
 	r.ReportCounter(name, tags, curr-prev)
 }
 
-func (g *gauge) Update(v int64) {
-	atomic.StoreInt64(&g.curr, v)
-	atomic.StoreInt64(&g.updated, 1)
+func (c *counter) snapshot() int64 {
+	return atomic.LoadInt64(&c.curr) - atomic.LoadInt64(&c.prev)
```

Comment:
Don't we need to reset prev here much like report? Otherwise we'll snapshot this every second and the count will keep increasing forever, instead of giving you a delta between snapshots.